src/OrchardCore.Modules/OrchardCore.Contents/Endpoints/Api/GetEndpoint.cs

```diff
@@ -42,6 +45,6 @@ private static async Task<IResult> HandleAsync(
             return httpContext.ChallengeOrForbid("Api");
         }
 
-        return TypedResults.Ok(contentItem);
+        return Results.Json(contentItem, options.Value.SerializerOptions);
```

Comment:
How about?

```c#
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;
using OrchardCore.ContentManagement;
using OrchardCore.Json;

namespace Microsoft.AspNetCore.Http;

public static class ContentItemResults
{
    public static IResult Json(
        ContentItem contentItem,
        HttpContext httpContext,
        string contentType = null,
        int? statusCode = null)
    {
        var options = httpContext.RequestServices.GetRequiredService<IOptions<DocumentJsonSerializerOptions>>();
        return Results.Json(contentItem, options.Value.SerializerOptions, contentType, statusCode);
    }
}
```

And also an API controller extension for the same.

Reply:
I rather not. Here you are recreating a new method that does the same thing as Result.Json by called it differently.

I would just use the asp version.  Unless we added it as extension to the HttpContext and auto resolve the options. But again, I rather not have an extension

Reply:
This exactly auto-resolves the option with the `HttpContext`, just keeps the feel of the original `Results.Json`. I'm OK with an extension method as well though.

The problem currently is that you have to copy-paste the configuration code around, and worse, have to know what to do and when to do it. A common method is easier to discover.

Reply:
We could use an output formatter to automatically select the correct JSON options. But that may be overkill for this purpose?

Reply:
@gvkries this maybe a good idea. Adding the following class may fix this problem. Also, users don't have to worry about this change since we'll just handle any document and entity for them even if they have custom API that returns document or entity.

If we have a problem repo, I can confirm this solution.

```
public class DocumentSystemTextJsonOutputFormatter : SystemTextJsonOutputFormatter
{
    public DocumentSystemTextJsonOutputFormatter(IOptions<DocumentJsonSerializerOptions> options)
        : base(options.Value.SerializerOptions)
    {
    }

    protected override bool CanWriteType(Type type)
    {
        return typeof(IDocument).IsAssignableFrom(type) ||
            typeof(IEntity).IsAssignableFrom(type);
    }
}
```

Reply:
A formatter is even better. Just we shouldn't have to copy around the same but crucial code.

Reply:
@Piedone yes but I don't think the formatter is going to apply in MiniumalAPI.

Reply:
Then we'd still also need some centralized code like discussed above.

Reply:
I like that extension. I rather keep the code int he 4 places as is for the MimilalAP unless we have a solution like the formatter approach. Reading `Results.Json(contentItem, options.Value.SerializerOptions)` in the minimalAPI will tell me exactly what is happening without having to worry about an extension that wrap one line of code.

Reply:
If you like that extension then why not use that?

Reply:
I meant to say I **don't** like that extension